llmc/rand.h

```diff
@@ -220,4 +220,21 @@ void normal_(float* data, unsigned int numel, float mean, float std, mt19937_sta
     }
 }
 
+void random_permutation_with_init(int* data, int numel, mt19937_state* state, int should_init) {
```

Comment:
this seems like a weird double-use function. I would say it is cleaner to separate init from shuffling

Reply:
agree, i was thinking how best to do it, and converged to this solution... since it's simple i thought it's ok